spacy/kb_base.pyx

```diff
@@ -0,0 +1,168 @@
+
+    DOCS: https://spacy.io/api/kb/#candidate_init
+    """
+
+    def __init__(self, BaseKnowledgeBase kb, entity_hash, entity_freq, entity_vector, alias_hash, prior_prob):
+        self.kb = kb
+        self.entity_hash = entity_hash
+        self.entity_freq = entity_freq
+        self.entity_vector = entity_vector
+        self.alias_hash = alias_hash
+        self.prior_prob = prior_prob
+
+    @property
+    def entity(self):
+        """RETURNS (uint64): hash of the entity's KB ID/name"""
+        return self.entity_hash
+
+    @property
+    def entity_(self):
+        """RETURNS (str): ID/name of this entity in the KB"""
+        return self.kb.vocab.strings[self.entity_hash]
+
+    @property
+    def alias(self):
+        """RETURNS (uint64): hash of the alias"""
+        return self.alias_hash
+
+    @property
+    def alias_(self):
+        """RETURNS (str): ID of the original alias"""
+        return self.kb.vocab.strings[self.alias_hash]
+
+    @property
+    def entity_freq(self):
+        return self.entity_freq
+
+    @property
+    def entity_vector(self):
+        return self.entity_vector
+
+    @property
+    def prior_prob(self):
+        return self.prior_prob
+
+
+def get_candidates_batch(kb: BaseKnowledgeBase, mentions: Iterable[Union[Span, str]]) -> Iterable[Iterator[Candidate]]:
+    """
+    Return candidate entities for the given mentions and fetching appropriate entries from the index.
+    kb (BaseKnowledgeBase): Knowledge base to query.
+    mention (Iterable[Union[Span, str]]): Entity mentions for which to identify candidates.
+    RETURNS (Iterable[Iterator[Candidate]]): Identified candidates.
+    """
+    return kb.get_candidates_batch(mentions)
+
+
+def get_candidates(kb: BaseKnowledgeBase, mention: Union[Span, str]) -> Iterator[Candidate]:
+    """
+    Return candidate entities for a given mention and fetching appropriate entries from the index.
+    kb (BaseKnowledgeBase): Knowledge base to query.
+    mention (Union[Span, str]): Entity mention for which to identify candidates.
+    RETURNS (Iterator[Candidate]): Identified candidates.
+    """
+    return kb.get_candidates(mention)
+
+
+cdef class BaseKnowledgeBase:
+    """A `BaseKnowledgeBase` instance stores unique identifiers for entities and their textual aliases,
+    to support entity linking of named entities to real-world concepts.
+    This is an abstract class and requires its operations to be implemented.
+
+    DOCS: https://spacy.io/api/kb
+    """
+
+    def __init__(self, vocab: Vocab, entity_vector_length: int):
+        """Create a KnowledgeBase."""
+        self.vocab = vocab
+        self._entity_vector_length = entity_vector_length
+        self.mem = Pool()
+
+    property entity_vector_length:
+        def __get__(self):
+            return self._entity_vector_length
+        def __set__(self, int value):
+            self._entity_vector_length = value
```

Comment:
The previous double-declaration/usage of `entity_vector_length` bugged me. If necessary, we can revert this - should be backwards-compatible though.